### PR TITLE
UI: Fix enableShortcuts, exclude it from being persisted

### LINKED ISSUE #6569: enableShortcuts not working in v5
**Describe the bug**
The `enableShortcuts` and `isFullscreen` options are not working as expected.

**To Reproduce**
Add this to `config.js` (skipping boilerplate)

```js
addParameters({
  options: {
    enableShortcuts: false,
    isFullScreen: true,
  },
})
```

**Expected behavior**
Shortcuts should be disabled.
Story should be fullscreen.

**System:**
Version: 5.x

### PR DESCRIPTION
**Pull request by ndelangen**
Closes https://github.com/storybookjs/storybook/issues/6569

## What I did

fix issue where if setConfig is called before channel is setup it no-ops
fix issue where setConfig would not call with full value
fix issue where state.ui is persisted where it should not

## How to test

- set `enableShortcuts: true` in `manager.js`, check if keyboardshortcuts work
- set `enableShortcuts: false` in `manager.js`, check if keyboardshortcuts are indeed no longer working
- set `enableShortcuts: true` in `manager.js`, check if keyboardshortcuts wokring again.

Between each step you will have to restart storybook.

### CODE REVIEW

File: code/lib/manager-api/src/modules/layout.ts [hunk 1]
```diff
@@ -253,7 +253,7 @@ export const init: ModuleFn = ({ store, provider, singleStory, fullAPI }) => {
     },
   };
 
-  const persisted = pick(store.getState(), 'layout', 'ui', 'selectedPanel');
+  const persisted = pick(store.getState(), 'layout', 'selectedPanel');
```
Comment (tmeasday) [inline code/lib/manager-api/src/modules/layout.ts:256]: Can you explain this change? Seems sort of wide ranging?

**Review by tmeasday — COMMENTED** [review]
The main part of this seems good. Very unsure about the persistence change though.

**Review by shilman — COMMENTED** [review]
@ndelangen Thanks for fixing this! The code looks OK for me.

However, as I was testing it, I found that I needed to do a hard reload in the browser after restarting Storybook, which still feels like a bug. Is there any way we can fix this?

I also tested a few of [the other UI options](https://storybook.js.org/docs/react/configure/features-and-behavior) and they don't work. Can open a separate issue if you think it's unrelated to this PR:

```
  isFullscreen: true,
  panelPosition: 'bottom',
  showRoots: false,
```

**Comment by ndelangen** [comment]
@tmeasday I found that anything you put into `manager.js` is basically going to get ignored forever if it's already persisted.
Which makes sense, for stuff you can change in storybook, but it makes no sense for the `state.ui` part, I think.

> However, as I was testing it, I found that I needed to do a hard reload in the browser after restarting Storybook, which still feels like a bug.

I assume when you say "restarting storybook" that includes a browser refresh? So you're saying you had to refresh twice to see the effect? That would be a bug, though I'm not sure what the problem is...

File: code/lib/manager-api/src/modules/layout.ts (hunk 1 above)
Comment (ndelangen) [reply code/lib/manager-api/src/modules/layout.ts:256]: https://github.com/storybookjs/storybook/blob/42fae30325bb7635e38689e8cc9ece3740d49746/code/lib/types/src/modules/api.ts#L105-L109

According to the types, these are the only options allowed in this object... And I think none of these should really be persisted!

**Comment by ndelangen** [comment]
@shilman what you found, is related to persistence, and it's an existing issue.

What happens is:
- upon first browser opening, `manager.js` is loaded, and it immediately sets persistence in localStorage.
- upon subsequent browser openings, `manager.js` is loaded, but it's values are overridden by the persisted data.
- if `manager.js` changes, this doesn't change anything about the above.

TLDR:
Once the data is persisted, It cannot be overridden anymore.

This PR, fixes that for `enableShortcuts` by not persisting the `.ui` property anymore.
Because `enableShortcuts` can't be changed in the UI anywhere at all, this seems like a good choice.

**Comment by ndelangen** [comment]
@shilman I've verified, that setting:

```ts
addons.setConfig({
  panelPosition: 'right',
});
```

...will work, only if you clear site data first.

I'm not sure what else to expect..
We could have a long discussion about exactly what should be persisted when, but it feels like a non-7.0 change?

**Review by shilman — APPROVED** [review]
Let's merge this and figure out a proper solution to this later
